Hardware/Hardware/CardReader/StripeCardReader/StripeCardReaderService.swift

```diff
@@ -503,7 +503,12 @@ extension StripeCardReaderService: BluetoothReaderDelegate {
                 error: CardReaderServiceError.softwareUpdate(underlyingError: UnderlyingError(with: error),
                                                              batteryLevel: reader.batteryLevel?.doubleValue))
             )
-            softwareUpdateSubject.send(.available)
+            if let requiredDate = update?.requiredAt,
+               requiredDate > Date() {
+                softwareUpdateSubject.send(.available)
+            } else {
+                softwareUpdateSubject.send(.none)
```

Comment:
❓ I am new to the Payments code, is this how we know a software update is required or optional - by comparing `ReaderSoftwareUpdate`'s `requiredAt` date to the current date? Why not including whether an update is required or optional in `softwareUpdateSubject`'s value like `.available(UpdateTypeEnum)` where the enum has two cases, required and optional?

Reply:
The canonical way is whether or not `.available` is sent by the Terminal SDK first, or if it just sends `started` and kicks the update off on connecting the reader. However, that's not useful after the required update has failed, so using the date is the next best thing, IMO